test.sh

```diff
@@ -0,0 +1,85 @@
+#!/bin/bash
+set -euo pipefail
+
+function cleanup {
+	if [[ -n "${test_dir+x}" ]]; then
+		rm -rf "$test_dir"
+	fi
+}
+
+# Register the cleanup function to run on exit or interrupt
+trap cleanup EXIT INT
+
+command="deploy --prod"
+site_url="https://dash.lagon.app"
+action_repo="lagonapp/github-action@main"
+function_path=
+
+# Parse inputs
+while getopts ":c:s:r:f:" opt; do
+	case $opt in
+	c)
+		# Optional set the `command` input
+		command="$OPTARG"
+		;;
+	s)
+		# Optionally set the `site_url` input
+		site_url="$OPTARG"
+		;;
+	r)
+		# This tells act which lagon-action to run
+		action_repo="$OPTARG"
+		;;
+	f)
+		# This is the folder that the action will run with (build and deploy)
+		function_path="$OPTARG"
+		;;
+	\?)
+		echo "Invalid option -$OPTARG" >&2
+		;;
+	esac
+done
+
+# Check that required options are set
+if [ -z "$function_path" ]; then
+	echo "Usage: $0 -f path_to_function [-r repo, -c command, -s site_url]" >&2
+	exit 1
+fi
+
+if ! [ -v LAGON_API_TOKEN ]; then
```

Comment:
I don't know a lot about bash, but I'm getting this error: `./test.sh: line 49: [: -v: unary operator expected` here. Shouldn't this line be:

```suggestion
if [ -z $LAGON_API_TOKEN ]; then
```

Reply:
`-z` just checks if the variable is empty but `-v` checks if the variable is unset. If the environment variable isn't set but we check if it's empty you get an unbounded error instead of the one I wanted to log.

Maybe it's a bsd (mac) vs linux bash difference... I've used a different check that is hopefully more portable.